build.gradle

```diff
@@ -701,19 +804,36 @@ def deobf(String sourceURL) {
 }
 
 // The method above is to be preferred. Use this method if the filename is not at the end of the URL.
-def deobf(String sourceURL, String fileName) {
+def deobf(String sourceURL, String rawFileName) {
+    String bon2Version = "2.5.1"
+    String fileName = URLDecoder.decode(rawFileName, "UTF-8")
     String cacheDir = System.getProperty("user.home") + "/.gradle/caches/"
     String bon2Dir = cacheDir + "forge_gradle/deobf"
-    String bon2File  = bon2Dir + "/BON2-2.5.0.jar"
+    String bon2File  = bon2Dir + "/BON2-" + bon2Version + ".jar"
     String obfFile = cacheDir + "modules-2/files-2.1/" + fileName + ".jar"
     String deobfFile = cacheDir + "modules-2/files-2.1/" + fileName + "-deobf.jar"
 
     if(file(deobfFile).exists()) {
         return files(deobfFile)
     }
 
+    checkPropertyExists("channel")
+    checkPropertyExists("mappingsVersion")
```

Comment:
This will most likely break all projects when `updateBuildScript` is used, can you add default values to these properties in build.gradle so that if they're missing from properties.gradle the script keeps on working?